nixos/modules/services/network-filesystems/ipfs.nix

```diff
@@ -128,7 +128,7 @@ in
         install -m 0755 -o ${cfg.user} -g ${cfg.group} -d ${cfg.dataDir}
       '';
       script =  ''
-        if [[ ! -d ${cfg.dataDir}/config ]]; then
+        if [[ ! -f ${cfg.dataDir}/config ]]; then
```

Comment:
My bash knowledge isn't great, so I didn't notice this before, `config` is a file, not a directory